File: samples/SampleModels/Models/Person.cs
using System.Collections.Generic;

namespace SampleModels.Models
{
    public class Person
    {
        public string Name { get; set; }
        public string Company { get; set; }
        public string Company4 { get; set; }

        [Optional]
        public string Company5 { get; set; }
        public int? Money { get; set; }
        public SubPerson Related { get; set; }
        public SubPerson Related2 { get; set; }
        public Person Related3 { get; set; }
        public List<SubPerson> Persons { get; set; }
        public SubPerson[] Persons2 { get; set; }
        public List<Person> Children { get; set; }
        public Person[] Children2 { get; set; }
        public List<string> Bills { get; set; }
        public string[] Bills2 { get; set; }
        public List<int> FavoriteNumbers { get; set; }
        public int[] FavoriteNumbers2 { get; set; }
        public GenericPerson<SubPerson> Data { get; set; }
        public GenericPerson<Person> Data2 { get; set; }
        public GenericPerson<GenericPerson<Person>> Data3 { get; set; }
        public List<List<Person>> People { get; set; }

        public List<List<string>> Strings { get; set; }
        public string[][] Strings2 { get; set; }
        public AccessLevel AccessLevel { get; set; }

        [TypeUnion("type1", "type2")]
        public string TypeUnions { get; set; }

        [TypeUnion(1.2, 2.2)]
        public double TypeUnionDouble { get; set; }

        public object ShouldBeAny { get; set; }

        [UnknownObject]
        [Optional]
        public object ShouldBeUnknown { get; set; }
    }
}

File: samples/SampleModels/Typescript/Person.ts
import { SubPerson } from "./SubPerson";
import { GenericPerson } from "./GenericPerson";
import { AccessLevel } from "./AccessLevel";

export interface Person {
   name: string;
   company: string;
   company4: string;
   money?: number;
   related: SubPerson;
   related2: SubPerson;
   related3: Person;
   persons: SubPerson[];
   persons2: SubPerson[];
   children: Person[];
   children2: Person[];
   bills: string[];
   bills2: string[];
   favoriteNumbers: number[];
   favoriteNumbers2: number[];
   data: GenericPerson<SubPerson>;
   data2: GenericPerson<Person>;
   data3: GenericPerson<GenericPerson<Person>>;
   people: Person[][];
   strings: string[][];
   strings2: string[][];
   accessLevel: AccessLevel;
   typeUnions: "type1" | "type2";
}
